packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import DrawingToolMarks from '../DrawingToolMarks'
+import SHOWN_MARKS from '@helpers/shownMarks'
+
+function PreviousMarks (props) {
+  const {
+    frame = 0,
+    interactionTaskAnnotations = [],
+    scale = 1,
+    shownMarks = 'ALL'
+  } = props
+  const marksToShow = shownMarks === SHOWN_MARKS.ALL || shownMarks === SHOWN_MARKS.USER
+
+  if (interactionTaskAnnotations?.length > 0 && marksToShow) {
+    // Wrapping the array in an react fragment because enzyme errors otherwise 
+    // React v16 allows this, though. 
+    return (
+      <>
+        {interactionTaskAnnotations.map((annotation) => {
+          const annotationValuesPerFrame = annotation.value.filter(value => value.frame === frame)
```

Comment:
@eatyourgreens this is a long-ago PR, but I have a question particularly about this line introduced here, which is still used in production. `annotationValuesPerFrame` is intended to filter marks based on the `frame` passed to PreviousMarks component. However, Travis and I have found that's not always the case and are unsure how long the following bug has been around ~

On projects with multi-frame subjects and more than one workflow step, previous annotations do not filter correctly when switching between frames. For instance, [How Did We Get Here](https://www.zooniverse.org/projects/communitiesandcrowds/how-did-we-get-here?demo=true) is a transcription project with drawing as the first step and a yes/no as the second step. If I draw marks on all the frames in the first step, while on the 2nd step navigating between frames renders marks on only _one_ frame. I think the expected behavior is that previous marks should always render on their respective frames. Any thoughts on how to fix this filter?

(More dicussion [here](https://github.com/zooniverse/front-end-monorepo/pull/5913#issuecomment-1945219415)).

Reply:
I can confirm the bug. Here, I've added two lines to frame 0 and one line to frame 1. `classification.annotations` only contains one line, not three.

Are annotations being lost on live projects that transcribe more than one page per subject?

<img width="800" alt="Screenshot of the dev classifier for How Did We Get Here. The console is open to show that classification.annotations only contains one transcribed line." src="https://github.com/zooniverse/front-end-monorepo/assets/59547/f2a728ac-ec8d-460a-9ba0-e60058f890da">

Reply:
When I go back to the drawing step, my drawn lines are there. They haven't been added to the classification. So my thought would be that the bug is in the code that adds mark IDs to the current classification.

<img width="1225" alt="Screenshot of the dev classifier for How Did We Get Here. There are two drawn lines on the page, but neither are stored in classification.annotations yet." src="https://github.com/zooniverse/front-end-monorepo/assets/59547/bee708ac-bada-49fc-86d9-0e521e2d0e52">

Reply:
If I then add a third mark to frame 0, `classification.annotations` updates to show the three lines from that frame, but loses the line from frame 1.

<img width="800" alt="Screenshot of the dev classifier for How Did We Get Here. I have added a third line to the first frame of the active subject, and `classification.annotations` now shows the three mark IDs for those lines, but has lost the ID of the fourth line (on the second frame.)" src="https://github.com/zooniverse/front-end-monorepo/assets/59547/a8313b24-ebf4-41bd-933f-e326c5309c5d">

Reply:
Based on that, I would say that `previousAnnotations` is working as expected, but the classification isn't being updated correctly when you draw on more than one frame of a subject. `classification.annotations` only contains mark IDs for the active frame. I'd expect it to contain the mark IDs for all the transcription lines that I've made.

Since this bug is in production, and there are projects with subjects that have more than one page, I would definitely double check that classifications are being recorded properly when a volunteer adds lines to more than one page.

Reply:
I've suggested a fix for the `markIDs` array in https://github.com/zooniverse/front-end-monorepo/pull/5919.

Reply:
Here is the bug, by the way. `marks` here is filtered by frame before being added to the classification in `InteractionLayer`.
https://github.com/zooniverse/front-end-monorepo/blob/6b421edae1e92322bf3676cc558612e886983321/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayerContainer.js#L92

Reply:
#5923 is a more robust fix, replacing the manual update with an autorun reaction whenever the task’s marks change.